core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesAction.java

```diff
@@ -228,9 +228,18 @@ public RewriteDataFilesActionResult execute() {
         .flatMap(Streams::stream)
         .collect(Collectors.toList());
 
+    // add a filter  to the CombinedScanTask list to avoid repeated rewrite datafile
+    List<CombinedScanTask> fileterCombinedScanTasks =
+        combinedScanTasks.stream().filter(task -> task.files().size() > 1).collect(Collectors.toList());
+
+    if (fileterCombinedScanTasks.isEmpty()) {
+      return RewriteDataFilesActionResult.empty();
+    }
+
     List<DataFile> addedDataFiles = rewriteDataForTasks(combinedScanTasks);
-    List<DataFile> currentDataFiles = filteredGroupedTasks.values().stream()
-        .flatMap(tasks -> tasks.stream().map(FileScanTask::file))
+    List<DataFile> currentDataFiles = fileterCombinedScanTasks.stream()
```

Comment:
I think the flatmap here is unecessary since we can flatMap within the map operation instead,

```
    filteredCombinedScanTasks.stream()
        .flatMap(task -> task.files().stream().map(FileScanTask::file))
        .collect(Collectors.toList())
```

Reply:
yes thanks for your suggestion，I update it